### PR TITLE
ci: Windows VHD pipeline tests

### PR DESCRIPTION
**Pull request by jsturtevant**
**Reason for Change**:

This PR unblocks the Windows VHD scripts by doing two things:

- Removes Azure VM agents from the sysprep process 
  - this fixes an issue where there was a missing DLL on recent upgrade of the Azure VM agents that caused sysprep to fail: `azure-arm: IMAGE_STATE_UNDEPLOYABLE - missing VMAgentDisabler.dll.`  The VM agent doesn't do anything during sysprep so it is ok to remove it. If the image is deployed to Azure as a VM the agent will redeployed. 
- Fixes the e2e tests during the VHD build using `USE_MANAGED_IDENTITY=false`

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: vhd/packer/sysprep.ps1 [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Stop and remove Azure Agents to enable use in Azure Stack
+# If deployed in azure vm the agents will be added to the vms at deployment time
```
Comment (AbelHu) [inline vhd/packer/sysprep.ps1:2]: Do you plan to stop DiagTrack?

File: vhd/packer/sysprep.ps1 [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+Stop-Service WindowsAzureGuestAgent
+Stop-Service WindowsAzureNetAgentSvc
+Stop-Service RdAgent
+# Stop-Service WindowsAzureTelemetryService
```
Comment (AbelHu) [inline vhd/packer/sysprep.ps1:6]: Why not stop and delete WindowsAzureTelemetryService?

File: vhd/packer/sysprep.ps1 (hunk 1 above)
Comment (jsturtevant) [reply vhd/packer/sysprep.ps1:2]: We only want to stop DiagTrack in the pause image which is being done in: https://github.com/kubernetes/kubernetes/pull/95950

File: vhd/packer/sysprep.ps1 (hunk 2 above)
Comment (jsturtevant) [reply vhd/packer/sysprep.ps1:6]: The WindowsAzureTelemetryService has been merged into the RD agent in latest version of the agents: https://docs.microsoft.com/en-us/azure/virtual-machines/troubleshooting/windows-azure-guest-agent

> Starting in version 2.7.41491.971 of the VM Guest Agent, the Telemetry component is included in the RDAgent service, Therefore, you might not see this Telemetry service listed in newly created VMs.

This was left over from earlier refactoring.  I'll remove it from the file

**Review by jackfrancis — APPROVED** [review]
/lgtm

File: vhd/packer/sysprep.ps1 [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Stop and remove Azure Agents to enable use in Azure Stack
```
Comment (marosset) [inline vhd/packer/sysprep.ps1:1]: Thanks for pulling this out into a separate script!

**Review by marosset — APPROVED** [review]
/LGTM

**Comment by marosset** [comment]
@AbelHu @mainred FYI

**Pull request merged**